src/Umbraco.Core/Telemetry/TelemetryService.cs

```diff
@@ -84,8 +84,9 @@ public bool TryGetTelemetryReportData(out TelemetryReportData? telemetryReportDa
 
             packages.Add(new PackageTelemetry
             {
+                Id = manifest.PackageId,
```

Comment:
The telemetry data endpoint would need to store this new value or otherwise at least prefer the ID (if set) over the name.

Currently the [`AlternatePackageNames` in the `umbraco-marketplace.json` file](https://docs.umbraco.com/umbraco-dxp/marketplace/listing-your-package#json-schema) is used to match the reported package names to a NuGet package that's listed on the Marketplace, but that's not a 1:1 mapping and also not guaranteed to be unique.

Reply:
This should use `IPackagingService.GetAllInstalledPackages()` instead, so both package migrations and manifest are reported in the telemetry data, similar to what's displayed in the backoffice 'Installed packages' overview...

But that would also require adding `AllowPackageTelemetry` to the `InstalledPackage` class and optionally allow specifying that in the package migration plan as well (or otherwise default to true and only allow opting-out by explicitly adding a package manifest). Any thoughts?

Reply:
The telemetry data will now also include packages that only have a package migration 👍🏻 You can only opt-out of package telemetry by specifying this in the package manifest (so you'd still have to create a manifest if you want to opt-out).